src/manifestoo/git.py

```diff
@@ -0,0 +1,56 @@
+import re
+import subprocess
+from pathlib import Path
+from typing import List
+
+
+class BranchNotFound(Exception):
+    pass
+
+
+def get_current_branch() -> str:
+    command_current = ["git", "rev-parse", "--abbrev-ref", "HEAD"]
+    return subprocess.check_output(command_current).strip().decode()
+
+
+def check_branch_exists(branch: str) -> None:
+    try:
+        command_check = ["git", "rev-parse", "--verify", branch]
+        subprocess.check_output(command_check)
+    except subprocess.CalledProcessError:
+        raise BranchNotFound(f"Cannot find branch {branch}. Aborting.")
+
+
+def _get_new_manifest_files(branch: str, current_branch: str) -> List[str]:
+    command_new = ["git", "diff", "--name-only", "--diff-filter", "A", branch, current_branch]
+    new_files = subprocess.check_output(command_new).decode().split()
+    return [f for f in new_files if f.endswith("/__manifest__.py")]
+
+
+def get_branch_new_addons(branch: str, addons_dirs: List[Path]) -> List[str]:
+    return _get_branch_addons_by(_get_new_manifest_files, branch, addons_dirs)
+
+
+def _get_modified_files(branch: str, current_branch: str) -> List[str]:
+    command_modified = ["git", "diff", "--name-only", branch, current_branch]
```

Comment:
Would it work too if we don't pass the current branch here ?

Reply:
Well, the first problem is that there was the debated change to 'main' as default made de facto standard by GitHub, so 'master' isn't even canonic anymore, even though it's certainly still more used. Even with that, many projects use 'dev' as main branch.
You can get the default branch on a given remote, the remote being canonically 'origin', but there is no guarantee either. Because this is intended to run on the central repo, it's also a bit weird to need to check the default remote.

All in all, it seems that makes a lot of assumptions to avoid giving a default that you can simply add to your command.

(The one you don't need to pass is actually current_branch because you can use HEAD, but we already don't ask for it (syntax would be a bit awkward too, whereas this is pretty explicit).)